memory/addressSpace.go

```diff
@@ -0,0 +1,46 @@
+package memory
+
+import (
+	"errors"
+)
+
+var errAddrSpaceOccupied = errors.New("address space occopied")
+
+type AddressSpace struct {
+	Name      string
+	Start     uint64
+	Size      uint32
+	Addresses []*AddressSpace
```

Comment:
Address vs. AddressSpace.
I would like to unify either of them.
Or perhaps children would be better.